src/Infrastructure/Util/src/ESMF_UtilTypes.F90

```diff
@@ -1930,6 +1931,36 @@ function ESMF_FileFormatNe(FileFormat1, FileFormat2)
                                  FileFormat2%fileformat)
 
 end function ESMF_FileFormatNe
+!------------------------------------------------------------------------------
+#undef  ESMF_METHOD
+#define ESMF_METHOD "ESMF_FileFormatAsString"
+subroutine ESMF_FileFormatAsString(String, FileFormat)
+  character(len=*), intent(out) :: String
+  type(ESMF_FileFormat_Flag), intent(in) :: FileFormat
+
+  if (FileFormat == ESMF_FILEFORMAT_UNKNOWN) then
+     String = 'ESMF_FILEFORMAT_UNKNOWN'
+  else if (FileFormat == ESMF_FILEFORMAT_VTK) then
+     String = 'ESMF_FILEFORMAT_VTK'
+  else if (FileFormat == ESMF_FILEFORMAT_SCRIP) then
+     String = 'ESMF_FILEFORMAT_SCRIP'
+  else if (FileFormat == ESMF_FILEFORMAT_ESMFMESH) then
+     String = 'ESMF_FILEFORMAT_ESMFMESH'
+  else if (FileFormat == ESMF_FILEFORMAT_ESMFGRID) then
+     String = 'ESMF_FILEFORMAT_ESMFGRID'
+  else if (FileFormat == ESMF_FILEFORMAT_UGRID) then
+     String = 'ESMF_FILEFORMAT_UGRID'
+  else if (FileFormat == ESMF_FILEFORMAT_CFGRID) then
+     ! Note that ESMF_FILEFORMAT_CFGRID is the same as ESMF_FILEFORMAT_GRIDSPEC
+     String = 'ESMF_FILEFORMAT_CFGRID/ESMF_FILEFORMAT_GRIDSPEC'
+  else if (FileFormat == ESMF_FILEFORMAT_MOSAIC) then
+     String = 'ESMF_FILEFORMAT_MOSAIC'
+  else if (FileFormat == ESMF_FILEFORMAT_TILE) then
+     String = 'ESMF_FILEFORMAT_TILE'
+  else
+     String = '(Unexpected ESMF_FILEFORMAT value)'
```

Comment:
Note that I don't abort for a bad value because I couldn't see a good way to do this from here, given that the subroutine is used in a `x = y` context (so no opportunity for a return code). Instead I give a particular string in that case. Let me know if there's a better way to handle this.